upf/action.py

```diff
@@ -114,9 +132,15 @@ def add_precondition(self, precondition: Union[FNode, 'upf.Fluent', ActionParame
         self._preconditions.append(precondition_exp)
 
     def add_effect(self, fluent: Union[FNode, 'upf.Fluent'],
-                   value: Union[FNode, 'upf.Fluent', 'upf.Object', ActionParameter, bool]):
+                   value: Union[FNode, 'upf.Fluent', 'upf.Object', ActionParameter, bool],
+                   condition: Union[FNode, BoolExpression] = True, kind: int = ASSIGNMENT):
         """Adds the given action effect."""
-        fluent_exp, value_exp = self._env.expression_manager.auto_promote(fluent, value)
+        fluent_exp, value_exp, condition_exp = self._env.expression_manager.auto_promote(fluent, value, condition)
+        if not (self._env.type_checker.get_type(condition_exp) == self._env.type_manager.BoolType()):
+            raise UPFTypeError('Action condition is not a Boolean condition!')
         if not self._env.type_checker.is_compatible_type(fluent_exp, value_exp):
             raise UPFTypeError('Action effect has not compatible types!')
-        self._effects.append((fluent_exp, value_exp))
+        if condition_exp == self._env.expression_manager.TRUE():
+            self._effects.append(Effect(fluent_exp, value_exp, condition = None, kind = kind))
```

Comment:
We shuold use the same default for the condition, `True` or `None`.

Reply:
You are totally right. For now I am using True so that if a simple tautology is given as a condition, it can be simplified to True and the effect becomes unconditional.